geps/gep-1619.md

```diff
@@ -157,49 +281,401 @@ Generally, the implementation API programs the dataplane API; however these two
+
+#### Traffic Splitting
+
+In scenarios involving traffic splitting, session persistence impacts load balancing done after routing, regardless if
+it is attached to a service or a route. When a persistent session is established and traffic splitting is configured
+across services, the persistence to a single backend should be maintained across services. Consequently, a persistent
+session takes precedence over traffic split weights when selecting a backend after route matching. It's important to
+note that session persistence does not impact the process of route matching. See [Expected API Behavior](#expected-api-behavior)
+for example use cases on traffic splitting.
+
+#### Cookie Attributes
+
+While the API is intended to be generic, as described in [API Granularity](#api-granularity), a majority of
+implementations will employ session persistence through cookies. Therefore, let's explore the possibilities of cookie
+configuration for these APIs.
+
+A cookie is composed of various attributes, each represented as key=value pairs. While some attributes may have optional
+values, the cookie name attribute is the only mandatory one, and the rest are considered optional.
+
+The cookie attributes defined by [RFC6265](https://www.rfc-editor.org/rfc/rfc6265#section-5.2) are:
+- Name=_value_
 - Expires=_date_
 - Max-Age=_number_
 - Domain=_domain_
 - Path=_path-value_
 - Secure
 - HttpOnly
 
-However, there are de-facto standard attributes that have been proposed as updates to [rfc6265](https://www.rfc-editor.org/rfc/rfc6265#section-5.2):
-- SameSite=_[Strict|Lax|None]_ ([Draft RFC Update](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-cookie-same-site-00))
-- Partitioned ([Draft RFC Update](https://www.ietf.org/archive/id/draft-cutler-httpbis-partitioned-cookies-01.html))
-
-**Session Cookie Strategy**
+Other cookie attributes not defined by RFC6265, but are captured in draft RFCs and could be considered de facto
+standards due to wide acceptance are:
+- SameSite=[Strict|Lax|None]
+- Partitioned
 
-Session cookie strategy is configuration specific to the creation of the cookie by the dataplane. These are not specific configuration parameters of the cookie itself, but instead the operation to take when inserting the cookie such as:
-- Rewriting an existing cookie value
-- Inserting a new cookie
-- Prefixing an existing cookie
-- Preserving an existing cookie value
-- Only adding cookies on specific HTTP methods (POST/GET)
+All cookie attributes are considered as opaque values in this spec and are to be determined by the individual
+implementations. However, certain cookie attributes require special consideration, which we will discuss in detail.
+
+##### Path
+
+The `Path` attribute should be automatically determined by implementors. When the `SessionPersistencePolicy` policy is
+attached to an xRoute, the implementor should interpret the path as configured on the xRoute. However, if the
```

Comment:
@costinm from the client perspective `/foo/...` and `/bar/...` appear as different destinations so the client may not expect to have session persistence preserved between them but I agree it is not clear what the user expectation is

Reply:
I'm struggling to understand (not just the comments, but also, even this design I wrote on how an implementation can interpret a `path` attribute out of a route).

Is the fundamental dilemma here that you can't specify two paths in the cookie `Path` attribute? Could there be two versions of the `set-cookie` based on the route path you take to the backend? Or can a single backend have only 1 cookie? Or am I misunderstanding the issue?

Reply:
Unfortunately I can't help - like in the previous section I am just asking for a clarification of the intended use of routes so I understand it, not suggesting changes or specific behaviors.

Reply:
@costinm 
> The fundamental question is what we want - IMO the user will expect a user to still go to the same backend ( sticky ) for both the /foo/ and /bar/ request ( if same service is selected as dest)
I don't think that's always assumed. Consider a single K8s Service that services two different gRPC services which leads to /Service1/methodX and /Service2/methodY paths which could have different persistency requirements.

Reply:
We also discussed this in our [meeting](https://docs.google.com/document/d/1DYYyxM-O_JXuLPyWijabUbcl1cRBSFpX-OFi6eSijs8/edit#heading=h.gb5npubb0ftd), and came to the conclusion that session persistence would not be preserved across these two routes targeting a single backend (because they have different path attributes).

I updated the GEP to add a paragraph clarifying this concept.